### PR TITLE
Fixes #25729 - don't auto publish puppet install dist

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
[test katello]

**Comment by beav** [comment]
[test katello] prior test run doesn't want to load, not sure if test failures were related to PR or not

**Comment by beav** [comment]
@jlsherrill do I need to regen my repos after applying this patch? I did a sync which failed (expected), then applied the patch and restarted, but it still fails.

File: test/glue/pulp/repository_test.rb [hunk 1]
```diff
@@ -310,30 +310,6 @@ def test_pulp_scratchpad_checksum_type
...
-
-    def teardown
-      delete_repo(@p_forge)
-    end
-
-    def test_generate_distributors
-      refute_nil @p_forge.find_distributor
-    end
-  end
-
```
Comment (jlsherrill) [inline test/glue/pulp/repository_test.rb]: This test was removed, as there is a more up to date one in test/services/katello/pulp/repository/puppet_test.rb

**Comment by beav** [comment]
per irc convo, we need an upgrade task for this. marking as waiting on contributor

**Comment by jlsherrill** [comment]
upgrade task added.

**Comment by beav** [comment]
@jlsherrill if I apply the patch and create a new puppet repo, it works.

It looks like `auto_publish` isn't getting updated when the rake task runs but I am not sure why. I'll `katello:reset` again and back out the patch and see if I can figure out whats happening

**Comment by beav** [comment]
upgrade job is successful, I see that `puppet_install_distributor` is updated to `auto_publish: false` but `puppet_distributor` is not.

**Comment by beav** [comment]
selection_028

ACK thanks @jlsherrill !

**Pull request merged**